server/src/main/java/io/crate/expression/scalar/AgeFunction.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar;
+
+import io.crate.data.Input;
+import io.crate.metadata.FunctionName;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.Signature;
+import io.crate.metadata.pgcatalog.PgCatalogSchemaInfo;
+import io.crate.types.DataTypes;
+import org.joda.time.Interval;
+import org.joda.time.Period;
+import org.joda.time.PeriodType;
+
+import static io.crate.metadata.functions.Signature.scalar;
+
+public class AgeFunction extends Scalar<Period, Object> {
+
+    private static final FunctionName NAME = new FunctionName(PgCatalogSchemaInfo.NAME, "age");
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(
+            scalar(
+                NAME,
+                DataTypes.TIMESTAMP.getTypeSignature(),
+                DataTypes.INTERVAL.getTypeSignature()
+            ).withFeatures(NO_FEATURES),
+            AgeFunction::new
+        );
+
+        module.register(
+            scalar(
+                NAME,
+                DataTypes.TIMESTAMP.getTypeSignature(),
+                DataTypes.TIMESTAMP.getTypeSignature(),
+                DataTypes.INTERVAL.getTypeSignature()
+            ).withFeatures(NO_FEATURES),
+            AgeFunction::new
+        );
+    }
+
+    private final Signature signature;
+    private final Signature boundSignature;
+
+    public AgeFunction(Signature signature, Signature boundSignature) {
+        this.signature = signature;
+        this.boundSignature = boundSignature;
+    }
+
+    @Override
+    public Signature signature() {
+        return signature;
+    }
+
+    @Override
+    public Signature boundSignature() {
+        return boundSignature;
+    }
+
+    @Override
+    public Period evaluate(TransactionContext txnCtx, NodeContext nodeContext, Input<Object>... args) {
+        /*
+         We cannot create Period from millis difference
+         since sometimes we need to treat different millis range as same interval.
+         Postgres treats age between first days of the sequential months as
+         '1 month' despite on possibly different 28, 30, 31 days/millis range.
+         */
+        assert args.length > 0 && args.length < 3 : "Invalid number of arguments";
+
+        var arg1 = args[0].value();
+        if (arg1 == null) {
+            return null;
+        }
+        if (args.length == 1) {
+            long curDateMillis = txnCtx.currentInstant().toEpochMilli();
+            curDateMillis = curDateMillis - curDateMillis % 86400000; // current_date at midnight, similar to CurrentDateFunction implementation.
+            return getPeriod(curDateMillis, (long) args[0].value());
```

Comment:
```suggestion
            return getPeriod(curDateMillis, arg1);
```

Should avoid multiple `value()` calls on the same input - if it were a function expression it would evaluate multiple times.

Reply:
Thx for catching,  missed that